java/clients/src/main/java/sleeper/clients/teardown/TerminateEMRServerlessApplications.java

```diff
@@ -104,6 +114,16 @@ private boolean allApplicationsTerminated() {
         return applicationsStillRunning == 0;
     }
 
+    private boolean allJobsCancelled(String applicationId) {
+        ListJobRunsResponse jobRunResponse = emrServerlessClient
+                    .listJobRuns(ListJobRunsRequest.builder().applicationId(applicationId).build());
+
+        long terminatedJobs = jobRunResponse.jobRuns().stream().filter(jobRun -> jobRun.state().equals(JobRunState.CANCELLED)).count();
+        long runningJobs = jobRunResponse.jobRuns().size() - terminatedJobs;
+        LOGGER.info("{} jobs are still cancelling for application {}", runningJobs, applicationId);
+        return runningJobs == 0;
```

Comment:
If some jobs failed, won't they still come back as failed here, meaning this would wait indefinitely?

Reply:
I have tested with failed, success, scheduled and running jobs and the teardown works